crates/wasmparser/src/readers/core/types.rs

```diff
@@ -771,6 +781,13 @@ pub struct SubType {
     pub structural_type: StructuralType,
 }
 
+impl SubType {
+    /// Unwrap a `FuncType` or panic.
+    pub fn unwrap_func(&self) -> &FuncType {
+        self.structural_type.unwrap_func()
+    }
```

Comment:
Should this perhaps also check that `is_final` is true or false? (I forget which it should check for)

Reply:
The default is `final == true` but there is nothing stopping one from defining a non-final function type, and I'd imagine this would still want to return the underlying function type.